sympy/physics/vector/tests/test_vector.py

```diff
@@ -169,3 +169,11 @@ def test_vector_simplify():
     test4 = ((-4 * x * y**2 - 2 * y**3 - 2 * x**2 * y) / (x + y)**2) * N.x
     test4 = test4.simplify()
     assert (test4 & N.x) == -2 * y
+
+
+def test_vector_evalf():
+    a, b = symbols('a b')
+    v = pi * A.x
+    assert sstr(v.evalf(2)) == '3.1*A.x'
+    v = pi * A.x + 5 * a * A.y - b * A.z
+    assert sstr(v.evalf(3)) == '3.14*A.x + 5.0*a*A.y - b*A.z'
```

Comment:
Is this the proper way to test this? It isn't clear to me why testing string matches is correct. Should you be testing if the correct objects are being returned. This is checking that indirectly.

Reply:
Yes , I think this is the right way . Same way is being used in evalf() tests. 

If this way is not used and the tests are asserted , then the assertion fails irrespective of it's correctness.

Reply:
Ok.

Reply:
String testing is not the proper way to test this. Other tests that use string testing should be changed. You can use srepr to get the proper representation of low precision floats:
```python
In [105]: pi                                                                                                                      
Out[105]: π

In [106]: pi.evalf(3)                                                                                                             
Out[106]: 3.14

In [107]: pi.evalf(3) == 3.14                                                                                                     
Out[107]: False

In [108]: srepr(pi.evalf(3))                                                                                                      
Out[108]: "Float('3.1416', precision=13)"

In [109]: pi.evalf(3)._prec                                                                                                       
Out[109]: 13

In [110]: pi.evalf(3) == Float(3.1416, precision=13)                                                                              
Out[110]: True
```
The best way is like:
```python
In [112]: pi.evalf(3) == Float('3.1416', 3)                                                                                       
Out[112]: True
```

Reply:
Thanks @oscarbenjamin for the tip! I figured there was something better.

Reply:
I'll update the tests!